control-plane-operator/controllers/hostedcontrolplane/cloud/aws/params.go

```diff
@@ -8,20 +8,24 @@ import (
 )
 
 type AWSParams struct {
-	Zone      string                 `json:"zone"`
-	VPC       string                 `json:"vpc"`
-	ClusterID string                 `json:"clusterID"`
-	SubnetID  string                 `json:"subnetID"`
-	OwnerRef  *metav1.OwnerReference `json:"ownerRef"`
+	Zone                 string                 `json:"zone"`
+	VPC                  string                 `json:"vpc"`
+	ClusterID            string                 `json:"clusterID"`
+	SubnetID             string                 `json:"subnetID"`
+	OwnerRef             *metav1.OwnerReference `json:"ownerRef"`
+	RoleARN              string                 `json:"roleARN"`
+	WebIdentityTokenFile string                 `json:"webIdentityTokenFile"`
 }
 
-func NewAWSParams(hcp *hyperv1.HostedControlPlane) *AWSParams {
+func NewAWSParams(hcp *hyperv1.HostedControlPlane, roleARN, webIdentityTokenFile string) *AWSParams {
 	if hcp.Spec.Platform.AWS == nil {
 		return nil
 	}
 	p := &AWSParams{
-		ClusterID: hcp.Spec.InfraID,
-		VPC:       hcp.Spec.Platform.AWS.CloudProviderConfig.VPC,
+		ClusterID:            hcp.Spec.InfraID,
+		VPC:                  hcp.Spec.Platform.AWS.CloudProviderConfig.VPC,
+		RoleARN:              roleARN,
+		WebIdentityTokenFile: webIdentityTokenFile,
```

Comment:
if we follow this approach https://github.com/kubernetes/cloud-provider-aws/pull/238 do we really need this?